security/pkg/pki/util/generate_cert.go

```diff
@@ -305,8 +305,7 @@ func genCertTemplateFromCSR(csr *x509.CertificateRequest, subjectIDs []string, t
 		ExtKeyUsage:           extKeyUsages,
 		IsCA:                  isCA,
 		BasicConstraintsValid: true,
-		ExtraExtensions:       exts,
-		SignatureAlgorithm:    csr.SignatureAlgorithm}, nil
+		ExtraExtensions:       exts}, nil
```

Comment:
I don't think removing the specification solves the issue.
Probably we need to explicitly specify the SignatureAlgorithm (use the strongest) based on the signer's private key:
Use SHA512WithRSA if signer private key is RSA
Use ECDSAWithSHA512 if signer private key is ECDSA
https://golang.org/src/crypto/x509/x509.go?s=5131:5158#L172
WDYT?

Reply:
I'm not an expert in this area. So please forgive me if my comment is not correct.

Could we just use the default signature algorithm in the x509.go?  https://golang.org/src/crypto/x509/x509.go?s=5131:5158#L1965

If the default algorithm in x509.go is not strong enough, should we let users choose the algorithm? Or we just specify the strongest algorithm here.

Reply:
Thanks. The default signature algorithm should be OK. Could we add one comment to make this clear?

Reply:
@cocotyty could you please add this comment?

Reply:
Sure, thanks.